django/urls/resolvers.py

```diff
@@ -631,11 +631,15 @@ def _reverse_with_prefix(self, lookup_view, _prefix, *args, **kwargs):
                     candidate_subs = kwargs
                 # Convert the candidate subs to text using Converter.to_url().
                 text_candidate_subs = {}
-                for k, v in candidate_subs.items():
-                    if k in converters:
-                        text_candidate_subs[k] = converters[k].to_url(v)
-                    else:
-                        text_candidate_subs[k] = str(v)
+                try:
+                    for k, v in candidate_subs.items():
+                        if k in converters:
+                            text_candidate_subs[k] = converters[k].to_url(v)
+                        else:
+                            text_candidate_subs[k] = str(v)
+                except ValueError:
+                    # Allow converters to decline to match by raising ValueError.
+                    continue
```

Comment:
This try-except could be a little tighter, and just go around `text_candidate_subs[k] = converters[k].to_url(v)`, but then we need some extra moving parts to continue from the nested `for` loop.

Reply:
Agreed, please move `try ... except` to the `text_candidate_subs[k] = converters[k].to_url(v)`.